docs/app/src/versions.js

```diff
@@ -2,36 +2,70 @@
 
 angular.module('versions', [])
 
-.controller('DocsVersionsCtrl', ['$scope', '$location', '$window', 'NG_VERSIONS', function($scope, $location, $window, NG_VERSIONS) {
+.controller('DocsVersionsCtrl', [
+  '$scope',
+  '$location',
+  '$window',
+  '$http',
+  '$q',
+  'NG_VERSIONS', function($scope, $location, $window, $http, $q, NG_VERSIONS) {
+
   $scope.docs_version  = NG_VERSIONS[0];
   $scope.docs_versions = NG_VERSIONS;
 
-  for (var i = 0, minor = NaN; i < NG_VERSIONS.length; i++) {
-    var version = NG_VERSIONS[i];
-    if (version.isSnapshot) {
-      version.isLatest = true;
-      continue;
-    }
-    // NaN will give false here
-    if (minor <= version.minor) {
-      continue;
+  // If this is not the snapshot version, request the snapshot's versions data
+  // to fill the version list with current data
+  $q(function(resolve, reject) {
+    if ($scope.docs_version.isSnapshot) {
+      reject();
+    } else {
+      resolve();
     }
-    version.isLatest = true;
-    minor = version.minor;
-  }
+  }).then(function() {
+    return $http.get('../snapshot/versions_data.json');
```

Comment:
Hmm. So for https://docs.angularjs.org (which is the snapshot) we cannot use this relative path. At the moment, it doesn't matter, because we only get the snapshot versions-data when we are not on the snapshot. 

For all other versions the root is https://code.angularjs.org/<version>/docs. In that case, this should be either `../../snapshot/versions_data.json` or `/snapshot/versions_data.json`. 

However, since we want to point docs.angularjs.org to the latest stable, an absolute url is probably the best idea. At the moment, this doesn't work because of CORS, though.